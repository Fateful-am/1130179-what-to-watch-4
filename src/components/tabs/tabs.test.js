import React from 'react';
import renderer from 'react-test-renderer';
import Tabs from './tabs';

describe(`Tabs component for MoviePage render correctly:`, () => {
  const mockClassNames = {
    list: `movie-nav__list`,
    item: `movie-nav__item`,
    activeItem: `movie-nav__item--active`,
    link: `movie-nav__link`
  };

  it(`with Overview tab`, () => {
    const tree = renderer
    .create(
        <Tabs
          tabs={[`Overview`, `Details`, `Reviews`]}
          className={mockClassNames}
          activeTab={`Overview`}
          onTabClick={()=>{}}
        />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`with Details tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            tabs={[`Overview`, `Details`, `Reviews`]}
            className={mockClassNames}
            activeTab={`Details`}
            onTabClick={()=>{}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`with Reviews tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            tabs={[`Overview`, `Details`, `Reviews`]}
            className={mockClassNames}
            activeTab={`Reviews`}
            onTabClick={()=>{}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe(`Tabs component for Main render correctly:`, () => {
  const mockClassNames = {
    list: `catalog__genres-list`,
    item: `catalog__genres-item`,
    activeItem: `catalog__genres-item--active`,
    link: `catalog__genres-link`
  };

  it(`with "All genres" tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            tabs={[`All genres`, `Comedy`, `Drama`]}
            className={mockClassNames}
            activeTab={`All genres`}
            onTabClick={()=>{}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`with "Comedy" tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            tabs={[`All genres`, `Comedy`, `Drama`]}
            className={mockClassNames}
            activeTab={`Comedy`}
            onTabClick={()=>{}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`with "Drama" tab`, () => {
    const tree = renderer
      .create(
          <Tabs
            tabs={[`All genres`, `Comedy`, `Drama`]}
            className={mockClassNames}
            activeTab={`Drama`}
            onTabClick={()=>{}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
