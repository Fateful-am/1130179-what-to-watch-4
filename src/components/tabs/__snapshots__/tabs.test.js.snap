// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component for Main render correctly: with "All genres" tab 1`] = `
<ul
  className="catalog__genres-list"
  onClick={[Function]}
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Drama
    </a>
  </li>
</ul>
`;

exports[`Tabs component for Main render correctly: with "Comedy" tab 1`] = `
<ul
  className="catalog__genres-list"
  onClick={[Function]}
>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Drama
    </a>
  </li>
</ul>
`;

exports[`Tabs component for Main render correctly: with "Drama" tab 1`] = `
<ul
  className="catalog__genres-list"
  onClick={[Function]}
>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item"
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
    >
      Drama
    </a>
  </li>
</ul>
`;

exports[`Tabs component for MoviePage render correctly: with Details tab 1`] = `
<ul
  className="movie-nav__list"
  onClick={[Function]}
>
  <li
    className="movie-nav__item"
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Overview
    </a>
  </li>
  <li
    className="movie-nav__item movie-nav__item--active"
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Details
    </a>
  </li>
  <li
    className="movie-nav__item"
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Reviews
    </a>
  </li>
</ul>
`;

exports[`Tabs component for MoviePage render correctly: with Overview tab 1`] = `
<ul
  className="movie-nav__list"
  onClick={[Function]}
>
  <li
    className="movie-nav__item movie-nav__item--active"
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Overview
    </a>
  </li>
  <li
    className="movie-nav__item"
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Details
    </a>
  </li>
  <li
    className="movie-nav__item"
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Reviews
    </a>
  </li>
</ul>
`;

exports[`Tabs component for MoviePage render correctly: with Reviews tab 1`] = `
<ul
  className="movie-nav__list"
  onClick={[Function]}
>
  <li
    className="movie-nav__item"
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Overview
    </a>
  </li>
  <li
    className="movie-nav__item"
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Details
    </a>
  </li>
  <li
    className="movie-nav__item movie-nav__item--active"
  >
    <a
      className="movie-nav__link"
      href="#"
    >
      Reviews
    </a>
  </li>
</ul>
`;
